Source/ARTLocalDevice.m

```diff
@@ -110,4 +110,23 @@ - (BOOL)isRegistered {
     return _identityTokenDetails != nil;
 }
 
+- (void)clearStorage {
+    for (NSString *key in @[ ARTDeviceIdKey, ARTDeviceIdentityTokenKey ]) {
+        [self.storage setObject:nil forKey:key];
+    }
+}
+
+- (void)reset {
+    self.id = @"";
+    self.secret = nil;
+    self.clientId = nil;
+    _identityTokenDetails = nil;
+    [self.push.recipient removeAllObjects];
+    [self clearStorage];
+}
+
+- (BOOL)isReset {
+    return [self.id isEqualToString:@""];
```

Comment:
@lawrence-forooghian I know this looks weird, but is a `nullable` `id` any better? Also additional flag, as I said earlier, probably also worse than this. Until the example app be able to handle `nil` device this is the best I could come up with. WDYT?

Reply:
Maybe using `extern NSString * const` as a `not set value` will be more elegant? You can even compare it using `==` (memory address)

Reply:
or `self.secret == nil` WDYT @lukasz-szyszkowski  @lawrence-forooghian @ben-xD

Reply:
for me `nil` is way better than the empty string

Reply:
I think my answer to this question probably depends on @maratal's answer to [this question](https://github.com/ably/ably-cocoa/pull/1253#discussion_r772363251).